File: src/main/java/org/ximo/finkinaction/java/datastream/connectors/KafkaConnectorConsumerApp.java
package org.ximo.finkinaction.java.datastream.connectors;

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;

import java.util.Properties;

/**
 * 连接阿里云的kafka解决
 * host.name=阿里云内网地址      #kafka绑定的interface
 * advertised.listeners=PLAINTEXT://阿里云外网映射地址:9092
 *
 * @author xikl
 * @date 2019/10/7
 */
public class KafkaConnectorConsumerApp {

    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        Properties properties = new Properties();
        properties.setProperty("bootstrap.servers", "47.98.200.49:9092");
        properties.setProperty("group.id", "test");

        String topic = "test";
        FlinkKafkaConsumer<String> flinkKafkaConsumer =
                new FlinkKafkaConsumer<>(topic, new SimpleStringSchema(), properties);

        final DataStreamSource<String> stringDataStreamSource = env.addSource(flinkKafkaConsumer);
        stringDataStreamSource.print();
        env.execute("KafkaConnectorConsumerApp-Java");

    }

}

File: src/main/scala/org/ximo/finkinaction/scala/datastream/connectors/KafkaConnectorConsumerApp.scala
package org.ximo.finkinaction.scala.datastream.connectors

import java.util.Properties

import org.apache.flink.api.common.serialization.SimpleStringSchema
import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer
import org.apache.flink.api.scala._

/**
  *
  * https://ci.apache.org/projects/flink/flink-docs-release-1.8/dev/connectors/kafka.html#kafka-consumers-start-position-configuration
  * @author xikl
  * @date 2019/10/9
  */
object KafkaConnectorConsumerApp extends App {

  val env = StreamExecutionEnvironment.getExecutionEnvironment

  val properties = new Properties()
  properties.setProperty("bootstrap.servers", "47.98.200.49:9092")
  properties.setProperty("group.id", "test")

  val topic = "test"

  val flinkKafkaConsumer = new FlinkKafkaConsumer[String](topic, new SimpleStringSchema(), properties)

  flinkKafkaConsumer.setStartFromGroupOffsets()

  val value: DataStream[String] = env.addSource(flinkKafkaConsumer)
  value.print()

  env.execute("KafkaConnectorConsumerApp-scala")



}
